src/core/cgroup.c

```diff
@@ -2449,7 +2472,7 @@ void unit_prune_cgroup(Unit *u) {
         if (is_root_slice)
                 return;
 
-        unit_release_cgroup(u);
+        (void) unit_maybe_release_cgroup(u);
```

Comment:
if the cgroup was not actually released the rest of the function should not be processed either I guess. Hence, taking my suggestion further up into account, maybe change this to:

```c
if (unit_maybe_release_cgroup(u) <= 0) /* Returns > 0 if the cgroup was pruned */
        return;
```